File: DesignPattern/ObserverPattern/src/main/java/MainObserver.java
/**
 * @project: observer-pattern
 * @author: HuyHuynh on 12/06/2021
 */
public class MainObserver {
    public static void main(String[] args) {
        CenterWeatherData centerWeatherData = new CenterWeatherData();
        CurrentDisplay currentDisplay = new CurrentDisplay(centerWeatherData);
        FutureDisplay futureDisplay = new FutureDisplay(centerWeatherData);

        centerWeatherData.measurementChanges(1,2,3);
        centerWeatherData.measurementChanges(3,4,5);
        centerWeatherData.measurementChanges(3,2,1);
        centerWeatherData.measurementChanges(5,6,7);
    }
}

File: DesignPattern/ObserverPattern/src/main/kotlin/MainObserver.kt
/**
 * @project: observer-pattern
 * @author: HuyHuynh on 12/06/2021
 */

fun main(){
    val dataCenterWeather = DataCenterWeather()
    val messageDisplay = MessageDisplay(dataCenterWeather)
    var currentDisplay = CurrentDisplayKotlin(dataCenterWeather)

    dataCenterWeather.measureDataChange(1F,2F,3F)
    dataCenterWeather.measureDataChange(4F,5F,6F)
    dataCenterWeather.remove(currentDisplay)
    dataCenterWeather.measureDataChange(7F,8F,9F)

}